hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -516,7 +516,7 @@ public void refreshSources(String peerId) throws IOException {
         ReplicationSourceInterface replicationSource = createSource(queueId, peer);
         this.oldsources.add(replicationSource);
         for (SortedSet<String> walsByGroup : walsByIdRecoveredQueues.get(queueId).values()) {
-          walsByGroup.forEach(wal -> src.enqueueLog(new Path(wal)));
+          walsByGroup.forEach(wal -> replicationSource.enqueueLog(new Path(wal)));
```

Comment:
So the problem is that logs from recovered queues were getting added to the new "normal" queue? And these logs are never read, then? 

Nit: maybe worth rename the variable from line #516 from `replicationSource` to `recoveredReplicationSource`, for further clarity?

I endorse the call for an additional UT, here.

Reply:
> So the problem is that logs from recovered queues were getting added to the new "normal" queue? And these logs are never read, then?

Yes. These log will be added to normal source. And will read by normal source. So this should not loss data but only affect the replicate order.

@ddupg I thought the issue title need to be changed.

Reply:
Thanks for clarifying, @infraio . Yes, let's change the title, please.

Reply:
Thank @wchevreuil @infraio for reviewing. In this case, it will loss data and duplicate data at the same time.

After we added these log to normal source, the normal source replicate the first WAL entries batch and updateLogPosition via zk.

It will successfully replicate first WAL entries batch, but data may be out-of date. And then failed to [updateLogPosition in zk](https://github.com/apache/hbase/blob/1164531d5ab519ab58af82ba3849f8fcded3453f/hbase-replication/src/main/java/org/apache/hadoop/hbase/replication/ZKReplicationQueueStorage.java#L233), which aborting RS, so we will also loss data.